### DISCUSSION TITLE
Custom Middleware

### DISCUSSION THREAD

**Discussion by BanDroid** [vikejs/vike#960] [Discussion] [answered]
since i'm new to this, where should i put my own middleware? before the viteDevMiddleware or after it?

**Comment by brillout** [comment]
https://github.com/brillout/vite-plugin-ssr/commit/cf6cf3e7320934b0626f239511476c3b228e4650

**Reply by BanDroid** [reply]
this does not work, i tried with express get and it return 404 not found, it always saying no route for /api page.

```javascript
app.get("/api", (req, res) => {
  res.status(200).json({ msg: "api called" });
});
```

**Reply by brillout** [reply]
I doubt it's related to VPS.

**Reply by BanDroid** [reply]
the project is still on localhost, any idea? should i enable `DEBUG=vps:routing`?

**Comment by BanDroid** [comment, marked as answer]
okay, i found out why.

it seems if i modify anything in server, i have to restart the server.
